.github/workflows/golangci-lint.yaml

```diff
@@ -8,37 +8,24 @@ on:
       - main
   pull_request:
 
+permissions:
+  contents: read
+
 jobs:
   golangci:
-    name: lint
-    runs-on: ubuntu-latest
+    name: Lint
+    runs-on: macos-latest
```

Comment:
@asmaloney may I ask why did you changed it to mac? Imo Ubuntu is faster

Reply:
I don't see that speed matter much here since the main build takes over 10 minutes (and I didn't notice much difference), but the Linux one was running an update and pulling down a bunch of packages. This is a waste of bandwidth and also means that it needs to be kept up-to-date if the required packages change.

Less required maintenance is good!

Reply:
Oh, you're right! Looks fair to me, thank you